src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEventArgs.cs

```diff
@@ -687,6 +692,7 @@ internal void FinishOperationSyncSuccess(int bytesTransferred, SocketFlags flags
                 LogBuffer(bytesTransferred);
             }
 
+            SocketType? socketType = _currentSocket?.SocketType;
```

Comment:
Why do we need to do this here rather than below where we log and only if the event source is enabled?  Is _currentSocket reset between here and there?

Reply:
Since the below switch block is quite big and `_currentSocket` is an instance field that can be changed in any method, I assumed it would be safer to read it here. Do you disagree?

Reply:
Maintainability is absolutely important.  This, however, is also a super hot path, and minimizing as much overhead as possible is more important.  If it's safe to move this into the logging if block, we should.

Reply:
Ok. Tomorrow, I will closely check all the branches to see if the read can be moved under `Log.IsEnabled()` check.

Reply:
`_currentSocket?.SocketType` call is moved down under `IsEnabled()` guarding check.